core/src/main/java/org/pentaho/di/core/Const.java

```diff
@@ -0,0 +1,4059 @@
+    }
+    return emptyPaddedSpacesStrings;
+  }
+
+  /**
+   * Return the percentage of free memory for this JVM.
+   *
+   * @return Percentage of free memory.
+   */
+  public static int getPercentageFreeMemory() {
+    Runtime runtime = Runtime.getRuntime();
+    long maxMemory = runtime.maxMemory();
+    long allocatedMemory = runtime.totalMemory();
+    long freeMemory = runtime.freeMemory();
+    long totalFreeMemory = ( freeMemory + ( maxMemory - allocatedMemory ) );
+
+    return (int) Math.round( 100 * (double) totalFreeMemory / maxMemory );
+  }
+
+  /**
+   * Return non digits only.
+   *
+   * @return non digits in a string.
+   */
+
+  public static String removeDigits( String input ) {
+    if ( Utils.isEmpty( input ) ) {
+      return null;
+    }
+    StringBuilder digitsOnly = new StringBuilder();
+    char c;
+    for ( int i = 0; i < input.length(); i++ ) {
+      c = input.charAt( i );
+      if ( !Character.isDigit( c ) ) {
+        digitsOnly.append( c );
+      }
+    }
+    return digitsOnly.toString();
+  }
+
+  /**
+   * Return digits only.
+   *
+   * @return digits in a string.
+   */
+  public static String getDigitsOnly( String input ) {
+    if ( Utils.isEmpty( input ) ) {
+      return null;
+    }
+    StringBuilder digitsOnly = new StringBuilder();
+    char c;
+    for ( int i = 0; i < input.length(); i++ ) {
+      c = input.charAt( i );
+      if ( Character.isDigit( c ) ) {
+        digitsOnly.append( c );
+      }
+    }
+    return digitsOnly.toString();
+  }
+
+  /**
+   * Remove time from a date.
+   *
+   * @return a date without hour.
+   */
+  public static Date removeTimeFromDate( Date input ) {
+    if ( input == null ) {
+      return null;
+    }
+    // Get an instance of the Calendar.
+    Calendar calendar = Calendar.getInstance();
+
+    // Make sure the calendar will not perform automatic correction.
+    calendar.setLenient( false );
+
+    // Set the time of the calendar to the given date.
+    calendar.setTime( input );
+
+    // Remove the hours, minutes, seconds and milliseconds.
+    calendar.set( Calendar.HOUR_OF_DAY, 0 );
+    calendar.set( Calendar.MINUTE, 0 );
+    calendar.set( Calendar.SECOND, 0 );
+    calendar.set( Calendar.MILLISECOND, 0 );
+
+    // Return the date again.
+    return calendar.getTime();
+  }
+
+  /**
+   * Escape XML content. i.e. replace characters with &values;
+   *
+   * @param content
+   *          content
+   * @return escaped content
+   */
+  public static String escapeXML( String content ) {
+    if ( Utils.isEmpty( content ) ) {
+      return content;
+    }
+    return StringEscapeUtils.escapeXml( content );
+  }
+
+  /**
+   * Escape HTML content. i.e. replace characters with &values;
+   *
+   * @param content
+   *          content
+   * @return escaped content
+   */
+  public static String escapeHtml( String content ) {
+    if ( Utils.isEmpty( content ) ) {
+      return content;
+    }
+    return StringEscapeUtils.escapeHtml( content );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

escapeJava(), escapeHtml3(), escapeHtml4(), escapeXml(), escapeXml10(), and escapeXml11() have been deprecated in org.apache.commons.lang.StringEscapeUtils and org.apache.commons.lang3.StringEscapeUtils and do not cover the entire range of characters that have to be encoded. OWASP Java Encoder library should be used instead.